### PR TITLE
Add application gateway backendHealthOnDemand operation

### PR DESCRIPTION
**Pull request by patoda**
Add backendhealthondemand operation for application gateway. It is an additive change, hence can go with existing API version.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by dsgouda** [comment]
@patoda Please target the network branch.
@MikhailTryakhov Please redirect @patoda and close this PR if necessary

**Comment by dsgouda** [comment]
@NelsonDaniel You can take a look at this

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json [hunk 1]
```diff
@@ -452,7 +452,71 @@
         "x-ms-long-running-operation": true
       }
     },
-	"/subscriptions/{subscriptionId}/providers/Microsoft.Network/applicationGatewayAvailableServerVariables": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationGateways/{applicationGatewayName}/backendHealthOnDemand": {
+      "post": {
+        "tags": [
+          "ApplicationGateways"
+        ],
+        "operationId": "ApplicationGateways_BackendHealthOnDemand",
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json]: This looks like a `GET` operation, any reason why it's a `POST` instead?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json [hunk 2]
```diff
@@ -692,6 +756,65 @@
     }
   },
   "definitions": {
+    "ApplicationGatewayOnDemandProbe": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json]: Please mark the readonly properties wherever applicable

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by patoda** [comment]
Targeting network branch and closing this one.
https://github.com/Azure/azure-rest-api-specs/pull/5552

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json (hunk 1 above)
Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json]: - Want to keep it similar to an exisiting operation - backendhealth which is a POST
- Have a request body
- Typically, we are using GET and PUT for getting and setting properties of a resource. For other operations on resource, we are using POST

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json (hunk 2 above)
Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/applicationGateway.json]: They are all writable in this case

**Pull request closed without merging**